gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java

```diff
@@ -61,15 +60,23 @@ public static int eip196_perform_operation(
     }
 
     if (ret != 0) {
-      var errBytes = Bytes.wrap(err);
-      err_len.setValue(errBytes.size() - errBytes.numberOfTrailingZeroBytes());
+      err_len.setValue(LibGnarkUtils.findFirstTrailingZeroIndex(err));
       o_len.setValue(0);
     } else {
       err_len.setValue(0);
     }
     return ret;
   }
 
+  public int findFirstTrailingZeroIndex(byte[] array) {
```

Comment:
this is implemented in the Utils class. I think this is not needed ...
You are even using the Utils implementation further up :-)

Reply:
where?  would love to DRY this out, but I don't see the usage you are talking about

Reply:
found and removed ;). thx

Reply:
This implementation has no usages, and can now be removed.